packages/flutter/test/material/icon_button_test.dart

```diff
@@ -3032,6 +3032,309 @@ void main() {
+    onLongPressed = false;
+
+    await tester.pumpWidget(build(enabled: false, onLongPress: onLongPress, onHover: onHover, useMaterial3: false));
+    await iconButtonFilledGesture.moveTo(iconButtonFilledOffset);
+    await tester.pumpAndSettle();
+    expect(onHovered, false);
+
+    await tester.longPressAt(iconButtonFilledOffset);
+    await tester.pumpAndSettle();
+    expect(onLongPressed, false);
+
+    await gesture.removePointer();
+
+    // IconButton.filledTonal
+    await tester.pumpWidget(build(onLongPress: onLongPress, onHover: onHover, useMaterial3: false));
+
+    final Finder iconButtonFilledTonal = find.widgetWithIcon(IconButton, Icons.add);
+    final Offset iconButtonFilledTonalOffset = tester.getCenter(iconButtonFilledTonal);
+    final TestGesture iconButtonFilledTonalGesture = await tester.createGesture(kind: PointerDeviceKind.mouse);
+    await iconButtonFilledTonalGesture.addPointer();
+
+    await iconButtonFilledTonalGesture.moveTo(iconButtonFilledTonalOffset);
+    await tester.pumpAndSettle();
+    expect(onHovered, true);
+
+    await tester.longPressAt(iconButtonFilledTonalOffset);
+    await tester.pumpAndSettle();
+    expect(onLongPressed, true);
+
+    onHovered = false;
+    onLongPressed = false;
+
+    await tester.pumpWidget(build(enabled: false, onLongPress: onLongPress, onHover: onHover, useMaterial3: false));
+    await iconButtonFilledTonalGesture.moveTo(iconButtonFilledTonalOffset);
+    await tester.pumpAndSettle();
+    expect(onHovered, false);
+
+    await tester.longPressAt(iconButtonFilledTonalOffset);
+    await tester.pumpAndSettle();
+    expect(onLongPressed, false);
+
+    await gesture.removePointer();
+
+    // IconButton.outlined
+    await tester.pumpWidget(build(onLongPress: onLongPress, onHover: onHover, useMaterial3: false));
+
+    final Finder iconButtonOutlined = find.widgetWithIcon(IconButton, Icons.add);
+    final Offset iconButtonOutlinedOffset = tester.getCenter(iconButtonOutlined);
+    final TestGesture iconButtonOutlinedGesture = await tester.createGesture(kind: PointerDeviceKind.mouse);
+    await iconButtonOutlinedGesture.addPointer();
+
+    await iconButtonOutlinedGesture.moveTo(iconButtonOutlinedOffset);
+    await tester.pumpAndSettle();
+    expect(onHovered, true);
+
+    await tester.longPressAt(iconButtonOutlinedOffset);
+    await tester.pumpAndSettle();
+    expect(onLongPressed, true);
+
+    onHovered = false;
+    onLongPressed = false;
+
+    await tester.pumpWidget(build(enabled: false, onLongPress: onLongPress, onHover: onHover, useMaterial3: false));
+    await iconButtonOutlinedGesture.moveTo(iconButtonOutlinedOffset);
+    await tester.pumpAndSettle();
+    expect(onHovered, false);
+
+    await tester.longPressAt(iconButtonOutlinedOffset);
+    await tester.pumpAndSettle();
+    expect(onLongPressed, false);
+  });
+}
+
+Widget build({bool enabled = true, bool useMaterial3 = true, void Function(bool)? onHover, VoidCallback? onLongPress}) {
```

Comment:
Consider renaming build with something more explicit as other tests in this file might need a helper function.
For instance, `buildAllVariants` or something better.